awx/main/managers.py

```diff
@@ -148,12 +145,15 @@ def register(self, uuid=None, hostname=None, ip_address=None, node_type='hybrid'
                     return (False, instance)
 
             # Create new instance, and fill in default values
-            create_defaults = dict(capacity=0, uuid=uuid)
+            create_defaults = dict(capacity=0)
             if defaults is not None:
                 create_defaults.update(defaults)
+            uuid_option = {}
+            if uuid is not None:
+                uuid_option = dict(uuid=uuid)
```

Comment:
you could just throw this into `create_defaults` as opposed to creating another dict. The double `**` seems to work, but I don't see any need for it here.

Reply:
I avoided `create_defaults` mostly because it wasn't technically a default. But if we're not too picky about that, then yep, sounds great.

Reply:
It's a default in the way that Django's [get_or_create](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#get-or-create) method referrers to defaults. If there's an existing object, the value will be ignored, but for a new object, it will be used.